### ISSUE TITLE
No matching global declaration available for the validation root.

### ISSUE THREAD

**Issue by lennerd** [sulu-io/sulu#1676]
When installing sulu as described in the current guides and opening the admin area, I get this error:

````
[ERROR 1845] Element '{http://friendsofsymfony.github.com/schema/rest}routes': No
matching global declaration available for the validation root. (in {project}/web/
- line 4, column 0) in {project}/vendor/sulu/sulu/src/Sulu/Bundle/SecurityBundle
/Resources/config/routing_api.xml (which is being imported from "{project}/app
/config/admin/routing.yml").
````
*I replaced the project base urls with `{project}`*

Looks like a XML validation error, but I'm not quite sure if this is a bug in the software or me.

Any ideas?

**Comment by lennerd** [comment]
I did a recent `composer update` which updated sulu to c50de07ca6f5e6d45926e4f0167c474537d57c2d. Still the same error.

**Comment by wachterjohannes** [comment]
do you have changed something in this routing file? `"{project}/app
/config/admin/routing.yml"` if yes could you please upload it here?

**Comment by lennerd** [comment]
Thank you for the quick response.

No, I haven't. But here is the corressponding `routing.yml`:

```yaml
sulu_tag:
    resource: "@SuluTagBundle/Resources/config/routing.yml"
    prefix:   /admin/tag

sulu_tag_api:
    resource: "@SuluTagBundle/Resources/config/routing_api.yml"
    prefix:   /admin/api

sulu_admin:
    resource: "@SuluAdminBundle/Resources/config/routing.yml"
    prefix: /admin

sulu_contact:
    resource: "@SuluContactBundle/Resources/config/routing.yml"
    prefix: /admin/contact

sulu_contact_api:
    type: rest
    resource: "@SuluContactBundle/Resources/config/routing_api.yml"
    prefix: /admin/api

sulu_security:
    resource: "@SuluSecurityBundle/Resources/config/routing.xml"
    prefix: /admin/security

sulu_security_api:
    type: rest
    resource: "@SuluSecurityBundle/Resources/config/routing_api.xml"
    prefix: /admin/api

sulu_content:
    resource: "@SuluContentBundle/Resources/config/routing.yml"
    prefix: /admin/content

sulu_content_api:
    type: rest
    resource: "@SuluContentBundle/Resources/config/routing_api.yml"
    prefix: /admin/api

sulu_media:
    resource: "@SuluMediaBundle/Resources/config/routing.yml"
    prefix: /admin/media

sulu_media_website:
    resource: "@SuluMediaBundle/Resources/config/routing_website.yml"

sulu_media_api:
    resource: "@SuluMediaBundle/Resources/config/routing_api.yml"
    prefix: /admin/api

sulu_category:
    resource: "@SuluCategoryBundle/Resources/config/routing.yml"
    prefix: /admin/category

sulu_category_api:
    resource: "@SuluCategoryBundle/Resources/config/routing_api.yml"
    prefix: /admin/api

sulu_snippet_api:
    resource: "@SuluSnippetBundle/Resources/config/routing_api.yml"
    type: rest
    prefix: /admin/api

sulu_location:
    resource: "@SuluLocationBundle/Resources/config/routing.yml"
    prefix: /admin/location

sulu_website:
    resource: "@SuluWebsiteBundle/Resources/config/routing.yml"
    prefix: /admin/website

sulu_websocket:
    type: rest
    resource: "@SuluWebsocketBundle/Resources/config/routing.yml"
    prefix: /admin/websocket

sulu_core:
    type: rest
    resource: "@SuluCoreBundle/Resources/config/routing_api.xml"
    prefix: /admin/api

# for preview
client_website:
    type: portal
    resource: "@ClientWebsiteBundle/Resources/config/routing.yml"

sulu_search:
    resource: "@SuluSearchBundle/Resources/config/routing.yml"
    prefix: /admin/search
```

**Comment by danrot** [comment]
I have seen this kind of error only if there is a wrong xml namespace in a `routing.xml` file, but that would mean that all installations would be affected, also the ones from us...

**Comment by lennerd** [comment]
Here is the corresponding XML file from the `SecurityBundle`:

```xml
<?xml version="1.0" encoding="UTF-8" ?>
<routes xmlns="http://friendsofsymfony.github.com/schema/rest"
        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
        xsi:schemaLocation="http://friendsofsymfony.github.com/schema/rest https://raw.github.com/FriendsOfSymfony/FOSRestBundle/master/Resources/config/schema/routing/rest_routing-1.0.xsd">
    <import id="sulu_security.roles" type="rest" resource="Sulu\Bundle\SecurityBundle\Controller\RoleController"/>
    <import id="sulu_security.groups" type="rest" resource="Sulu\Bundle\SecurityBundle\Controller\GroupController"/>
    <import id="sulu_security.users" type="rest" resource="Sulu\Bundle\SecurityBundle\Controller\UserController"/>
    <import id="sulu_security.permissions" type="rest" resource="sulu_security.permission_controller"/>
</routes>
```

Thanks again for your help! I'm really impressed by your work on Sulu so far.

**Comment by lennerd** [comment]
Which version of PHP do you use for your installations?

**Comment by wachterjohannes** [comment]
i think i have seen this with different xml lib versions (my is libxml2: stable 2.9.2 (bottled), HEAD) what did you have?

PHP Version: PHP 5.6.14 (cli) (built: Oct  5 2015 22:14:43)

**Comment by lennerd** [comment]
We are on an exact same state:

> libxml2 Version => 2.9.2
> PHP 5.6.14 (cli) (built: Oct  9 2015 20:41:53) 

I'm confused.

**Comment by lennerd** [comment]
FYI: I just ran the tests for the `FOSRestBundle`. There I got the same error multiple times. So this seems to be somehow related to the REST bundle. I have the feeling that something is broken in my `libxml` installation. If you have any more ideas for me ...

**Comment by lennerd** [comment]
:expressionless: https://github.com/FriendsOfSymfony/FOSRestBundle/pull/1169

**Comment by wachterjohannes** [comment]
let us know when we can update the bundle (: thanks for research and we hope that we can help next time!

**Comment by lennerd** [comment]
The bundle can be updated.

**Comment by wachterjohannes** [comment]
there is no new version in the https://github.com/FriendsOfSymfony/FOSRestBundle we are on version 1.7.1 ?

**Comment by lennerd** [comment]
It was merged in the 1.7 branch. Maybe there is no 1.7.2 tag yet.

**Comment by wachterjohannes** [comment]
ah ok i think if it is only a small release it will be done with 1.1 release soon (:
